prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -320,35 +321,49 @@ void rxr_pkt_req_data_from_tx(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry
 		return;
 	}
 
+	base_hdr = rxr_get_base_hdr(pkt_entry->pkt);
+	/*
+	 * For eager msg, we always copy data to packet entry.
```

Comment:
This comment is confusing.  In the zero copy case, we're still sending eager sized messages, but we don't need to copy.

I'd also rephrase this comment.  Because we've copied to deal with memory registration, we can opportunistically return completion earlier for performance gains.  We shouldn't copy just to return early; that's what the inject semantic is really all about.

Reply:
This patch actually eliminated zero copy send for eager messages.

The reason is to use zero copy send, we will post application's buffer to device (by calling `ibv_post_send`). In that case, we will need to wait for rdma-core's completion. Because `ibv_post_send` require caller to not alter buffer before receive cq entry.

An alternative is to set a flag in tx_entry indicating whether data has been copied to bounce buffer, and act accordingly in `rxr_pkt_handle_eager_rtm_sent`.

Reply:
In newest revision, I changed the logic back to avoid copy if possible. 
I added a flag to indicate whether zero copy send is being used.

If zero copy send is used, we do not write tx completion early.

Reply:
Removed this patch in new revision.